backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/textfield/internal/BpkTextFieldImpl.kt

```diff
@@ -0,0 +1,242 @@
+                        status is BpkFieldStatus.Error -> BpkTheme.colors.textError
+                        isFocused -> BpkTheme.colors.coreAccent
+                        else -> BpkTheme.colors.line
+                    },
+                ).value,
+            )
+            .background(BpkTheme.colors.surfaceDefault, Shape)
+            .padding(horizontal = BpkSpacing.Md),
+    ) {
+
+        if (icon != null) {
+            BpkIcon(
+                icon = icon,
+                contentDescription = null,
+                size = BpkIconSize.Large,
+                modifier = Modifier.padding(start = BpkSpacing.Sm),
+                tint = animateColorAsState(
+                    when (status) {
+                        is BpkFieldStatus.Disabled -> BpkTheme.colors.textDisabled
+                        else -> BpkTheme.colors.textSecondary
+                    },
+                ).value,
+            )
+        }
+
+        Box(
+            modifier = Modifier
+                .weight(1f)
+                .padding(BpkSpacing.Md),
+        ) {
+
+            BpkText(
+                text = placeholder ?: "",
+                color = BpkTheme.colors.textDisabled,
+                maxLines = maxLines,
+                modifier = Modifier.hideContentIf(value.text.isNotEmpty()),
+                style = BpkTheme.typography.bodyDefault,
+                overflow = TextOverflow.Ellipsis,
+            )
+
+            BasicTextField(
+                value = value,
+                onValueChange = onValueChange,
+                enabled = status != BpkFieldStatus.Disabled,
+                readOnly = readOnly,
+                textStyle = BpkTheme.typography.bodyDefault.copy(
+                    color = animateColorAsState(
+                        when (status) {
+                            is BpkFieldStatus.Disabled -> BpkTheme.colors.textDisabled
+                            else -> BpkTheme.colors.textPrimary
+                        },
+                    ).value,
+                ),
+                keyboardOptions = keyboardOptions,
+                keyboardActions = keyboardActions,
+                singleLine = maxLines == 1,
+                maxLines = maxLines,
+                visualTransformation = visualTransformation,
+                interactionSource = interactionSource,
+                cursorBrush = SolidColor(BpkTheme.colors.coreAccent),
+            )
+        }
+
+        var lastIcon by remember { mutableStateOf<Pair<BpkIcon, Color>?>(null) }
+        if (trailingIcon != null) {
+            BpkIcon(
+                icon = trailingIcon,
+                contentDescription = null,
+                size = BpkIconSize.Small,
+                tint = animateColorAsState(
+                    when (status) {
+                        is BpkFieldStatus.Disabled -> BpkTheme.colors.textDisabled
+                        else -> BpkTheme.colors.textSecondary
```

Comment:
Looking at figma this should be `textPrimary`

Reply:
good catch, thanks.